web/Services/AzureStorage.cs

```diff
@@ -0,0 +1,124 @@
+﻿using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+
+namespace BricksAndHearts.Services
+{
+    public interface IAzureStorage
+    {
+        Task CreateContainerAsync(string varType, int id);
+        Task DeleteContainerAsync(string varType, int id);
+        Task UploadFileAsync(IFormFile file, string varType, int id);
+        Task<List<string>> ListFilesAsync(string varType, int id);
+        Task<(Stream data, string fileType)> DownloadFileAsync(string varType, int id, string blobFilename);
+        Task DeleteFileAsync(string varType, int id, string blobFilename);
+    }
+
+    public class AzureStorage : IAzureStorage
+    {
+        private readonly string _storageConnectionString;
+        private readonly string _baseContainerName;
+        private readonly ILogger<AzureStorage> _logger;
+
+        public AzureStorage(IConfiguration configuration, ILogger<AzureStorage> logger)
+        {
+            _storageConnectionString = configuration.GetValue<string>("BlobConnectionString");
+            _baseContainerName = configuration.GetValue<string>("BlobContainerName");
+            _logger = logger;
+        }
+
+        private string GetContainerName(string varType, int id)
+        {
+            return _baseContainerName + varType + id.ToString();
+        }
+
+        private async Task<BlobContainerClient> GetContainerClient(string varType, int id)
+        {
+            string containerName = GetContainerName(varType, id);
+            BlobContainerClient containerClient = new BlobContainerClient(_storageConnectionString, containerName);
+            if (!containerClient.Exists())
+            {
+                await CreateContainerAsync(varType, id);
+                containerClient = new BlobContainerClient(_storageConnectionString, containerName);
+            }
+            return containerClient;
+        }
+
+        public async Task CreateContainerAsync(string varType, int id)

```

Comment:
Can be private?

Reply:
Indeed - there was a stage when a container was created as part of property creation, thus it had to be public, but in order to avoid problems with non-existent containers that changed and so the method can now be private.